[Feature] redis에 신호등 정보 저장 및 조회 기능 추가

ByungilOh-Fillip opened the pull request:
## #️⃣ 연관된 이슈
<!-- ex) #이슈번호[, #이슈번호] -->

#25 - Redis에 신호등 정보 캐싱하기


## ✅ 체크리스트

- [x] 🔀 PR 제목의 형식 맞추기 <!-- ex) [Feature] ㅇㅇ 기능 추가 -->
- [x] 💡 이슈 등록
- [x] 🏷️ 라벨 등록
- [x] 🧹 불필요한 코드 제거
- [x] 🧪 로컬 테스트 완료
- [x] 🏗️ 빌드 성공
- [x] 💯 테스트 통과


## 📝 작업 내용
/api/unifiedSignal/save - DB에 저장된 교차로와 보행등 정보를 redis에 저장
/api/unifiedSignal/find-info/crossroad/{id} - redis에 저장된 교차로 정보 조회
/api/unifiedSignal/find-info/traffic/{id} - redis에 저장된 보행등 정보 조회

## 📸 스크린샷 (UI 변경 시 필수)




## 👀 리뷰어 가이드라인
<!-- 리뷰어가 중점적으로 확인해야 할 사항을 작성해 주세요. -->
<!-- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->

작업 내용에 적힌 API 위주로 확인해주시면 될거 같습니다.
이전 리뷰해주신 사항 적용해두었습니다!

ByungilOh-Fillip committed:
feat: 신호등 위치정보 캐싱처리 및 id로 호출

ByungilOh-Fillip committed:
feat: 신호등 위치정보 캐싱처리 및 id로 호출

# Conflicts:
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CrossroadRedisRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CustomCrossroadRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CustomCrossroadRepositoryImpl.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/service/CrossroadService.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/entity/TrafficSignal.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepositoryImpl.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/TrafficRedisRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/service/TrafficService.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/unifiedSignal/controller/SignalController.java

ByungilOh-Fillip committed:
feat: 신호등 위치정보 캐싱처리 및 id로 호출

ByungilOh-Fillip committed:
feat: 신호등 위치정보 캐싱처리 및 id로 호출

# Conflicts:
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CrossroadRedisRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CustomCrossroadRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CustomCrossroadRepositoryImpl.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/service/CrossroadService.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/entity/TrafficSignal.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepositoryImpl.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/TrafficRedisRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/service/TrafficService.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/unifiedSignal/controller/SignalController.java

ByungilOh-Fillip committed:
feat: 신호등 위치정보 캐싱처리 및 id로 호출

ByungilOh-Fillip committed:
feat: 신호등 위치정보 캐싱처리 및 id로 호출

ByungilOh-Fillip committed:
feat: 신호등 위치정보 캐싱처리 및 id로 호출

ByungilOh-Fillip committed:
Test: SignalControllerTest 및 RestDoc 작성

ByungilOh-Fillip committed:
Test: SignalControllerTest 및 RestDoc 작성

ByungilOh-Fillip committed:
Test: SignalControllerTest 및 RestDoc 작성

ByungilOh-Fillip committed:
Test: SignalControllerTest 및 RestDoc 작성

ByungilOh-Fillip committed:
Test: SignalControllerTest 및 RestDoc 작성

ByungilOh-Fillip committed:
Test: SignalControllerTest 및 RestDoc 작성

ByungilOh-Fillip committed:
fix: error 해결 dto 생성자 작성

ByungilOh-Fillip committed:
fix: error 해결 dto 생성자 작성

ByungilOh-Fillip committed:
fix: error 해결 dto 생성자 작성

ByungilOh-Fillip committed:
fix: error 해결 dto 생성자 작성

ByungilOh-Fillip committed:
fix: error 해결 dto 생성자 작성

ByungilOh-Fillip committed:
fix: error 해결 dto 생성자 작성

ByungilOh-Fillip committed:
fix: error 해결 dto 생성자 작성

ByungilOh-Fillip committed:
refactor: RestDocs 파라미터값 제한 없게 변경

ByungilOh-Fillip committed:
refactor: RestDocs 파라미터값 제한 없게 변경

ByungilOh-Fillip committed:
refactor: RestDocs 파라미터값 제한 없게 변경

ByungilOh-Fillip committed:
refactor: RestDocs 파라미터값 제한 없게 변경

ByungilOh-Fillip committed:
refactor: RestDocs 파라미터값 제한 없게 변경

ByungilOh-Fillip committed:
refactor: RestDocs 파라미터값 제한 없게 변경

DongminL commented on src/main/java/org/programmers/signalbuddyfinal/domain/unifiedSignal/controller/SignalController.java:
```diff
@@ -0,0 +1,51 @@
+    private final CrossroadService crossroadService;
+    private final TrafficService trafficService;
+
+    @PostMapping("/save")
```
보통 캐시는 조회할 때 서버에서 자동으로 캐싱 데이터를 저장시키는데 API를 따로 만드신 이유가 있으신가요?

DongminL replied:
해당 API는 사용할 계획인가요?

ByungilOh-Fillip replied:
네 API 명을 변경해서 다시 올리려고하는데 
사용자 주변 데이터를 조회하고 redis에 저장하는 API입니다!

혹시 불필요한가요..?

DongminL replied:
아하 아닙니다! 그러면 api url을 변경하고, HTTP Method를 GET으로 하는게 좋을 것 같습니다!

ByungilOh-Fillip replied:
네! 안그래도 다시보다가 값을 받아야할거 같아서 수정중이였습니다!

ByungilOh-Fillip commented on src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CustomCrossroadRepositoryImpl.java:
```diff
@@ -45,6 +47,24 @@ public List<CrossroadApiResponse> findNearByCrossroads(double latitude, double l
+        }
+
+        return crossroadRes;
+    }
```
네 확인하고 통합 진행해보겠습니다!

ByungilOh-Fillip commented on src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/service/CrossroadService.java:
```diff
@@ -56,6 +62,34 @@ public void saveCrossroadDates(int page, int pageSize) {
+
+    public CrossroadResponse crossraodFindById(String id) {
+
+        if( !id.startsWith(KEY_PREFIX)){
+            id = KEY_PREFIX+id;
+        }
```
네 그렇네요 코드 정리 한번 해보겠습니다!

ByungilOh-Fillip commented on src/main/java/org/programmers/signalbuddyfinal/domain/unifiedSignal/controller/SignalController.java:
```diff
@@ -0,0 +1,51 @@
+
+    @GetMapping("find-info/crossroad/{id}")
+    public ResponseEntity<ApiResponse<Object>> findCrossInfo(
+            @PathVariable String id
```
아 잔여시간 검색하는 API랑 일치 시키려고 API_ID로 스웨거에는 명시해놨는데 여기도 변수명 맞추어두겠습니다!

ByungilOh-Fillip commented on src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/service/CrossroadService.java:
```diff
@@ -56,6 +62,34 @@ public void saveCrossroadDates(int page, int pageSize) {
+
+        CrossroadResponse response = crossroadRedisRepository.findById(id);
+
+        if(response.getCrossroadId() == null){
+            throw new BusinessException(CrossroadErrorCode.NOT_FOUND_CROSSROAD);
+        }
```
네! 감사합니다

ByungilOh-Fillip commented:
리뷰 반영 체크리스트 
- [ ] KEY_PREFIX 사용 정리
- [ ] findNearestCrossroad로 조회 기능 통합
- [ ] 캐싱 처리 순서  Read-Through 캐싱 반영

ByungilOh-Fillip commented:
> 리뷰 반영 체크리스트
> 
> * [ ]  KEY_PREFIX 사용 정리
> * [ ]  findNearestCrossroad로 조회 기능 통합
> * [ ]  캐싱 처리 순서  Read-Through 캐싱 반영

피드백 감사합니다! 반영 후 체크리스트에 표시해두겠습니다!

DongminL committed:
[Feature] 알림 사용자 기기 복수 등록 (#144)

* chore: Jenkins 대기 시간 늘림

* feat: 하나의 계정에서 디바이스 토큰 여러개 저장할 수 있게 함

* fix: 불필요한 api 제거

* feat: multicast 전송으로 변경

ByungilOh-Fillip committed:
refactor: 사용자 주변 신호등 보행등 검색 메소드 통일

ByungilOh-Fillip committed:
refactor: 사용자 주변 신호등 보행등 검색 메소드 통일

ByungilOh-Fillip committed:
refactor: 사용자 주변 신호등 보행등 검색 메소드 통일

ByungilOh-Fillip committed:
refactor: 사용자 주변 신호등 보행등 검색 메소드 통일

ByungilOh-Fillip committed:
refactor: 사용자 주변 신호등 보행등 검색 메소드 통일

ByungilOh-Fillip committed:
refactor: 사용자 주변 신호등 보행등 검색 메소드 통일

ByungilOh-Fillip committed:
refactor: KEY_PREFIC 정리

ByungilOh-Fillip committed:
refactor: KEY_PREFIC 정리

ByungilOh-Fillip committed:
refactor: KEY_PREFIC 정리

ByungilOh-Fillip committed:
refactor: KEY_PREFIC 정리

ByungilOh-Fillip committed:
refactor: KEY_PREFIC 정리

ByungilOh-Fillip committed:
refactor: Read-Through 캐싱 반영

ByungilOh-Fillip committed:
refactor: Read-Through 캐싱 반영

ByungilOh-Fillip committed:
refactor: Read-Through 캐싱 반영

ByungilOh-Fillip committed:
refactor: Read-Through 캐싱 반영

ByungilOh-Fillip committed:
refactor: 리뷰 적용 후 test 코드 수정

ByungilOh-Fillip committed:
refactor: 리뷰 적용 후 test 코드 수정

ByungilOh-Fillip committed:
refactor: 리뷰 적용 후 test 코드 수정

ByungilOh-Fillip committed:
refactor: 리뷰 적용 후 test 코드 수정

ByungilOh-Fillip committed:
chore: 코드 정리 쓰이지 않는 import 제거

ByungilOh-Fillip committed:
chore: 코드 정리 쓰이지 않는 import 제거

ByungilOh-Fillip committed:
chore: 코드 정리 쓰이지 않는 import 제거

ByungilOh-Fillip committed:
chore: 코드 정리 쓰이지 않는 import 제거

ByungilOh-Fillip committed:
chore: 코드 정리 쓰이지 않는 import 제거

ByungilOh-Fillip committed:
chore: 코드 정리 쓰이지 않는 import 제거

ByungilOh-Fillip committed:
Merge remote-tracking branch 'origin/feature/25-메인-redis에-신호등-정보-저장하기' into feature/25-메인-redis에-신호등-정보-저장하기

# Conflicts:
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CrossroadRedisRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CustomCrossroadRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CustomCrossroadRepositoryImpl.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/service/CrossroadService.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepositoryImpl.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/TrafficRedisRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/service/TrafficService.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/unifiedSignal/controller/SignalController.java
#	src/test/java/org/programmers/signalbuddyfinal/domain/unifiedSignal/controller/SignalControllerTest.java

ByungilOh-Fillip commented:
> > 리뷰 반영 체크리스트
> > 
> > * [x]   KEY_PREFIX 사용 정리
> > * [x]   findNearestCrossroad로 조회 기능 통합
> > * [x]   캐싱 처리 순서  Read-Through 캐싱 반영
> > * [x]   Controller 변수명 변경
> 
> 피드백 감사합니다! 반영 후 체크리스트에 표시해두겠습니다!

변경 완료했습니다!

DongminL commented on src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/service/CrossroadService.java:
```diff
@@ -56,6 +61,45 @@ public void saveCrossroadDates(int page, int pageSize) {
+                CrossroadResponse responseDB = new CrossroadResponse(
+                    crossroadRepository.findByCrossroadApiId( String.valueOf(apiId) ) );
+
+                crossroadRedisRepository.save(responseDB);
+                responseRedis = crossroadRedisRepository.findById(apiId);
```
responseDB의 값과 redis에 저장하고 `findById`로 가져오는 값이 동일해서, `findById`로 다시 가져오는 건 불필요해 보입니다!

ByungilOh-Fillip replied:
엇 그러면 사용자가 저장된 데이터를 다시 요청해야하나요? 
아니면 이미 redisTemple에서 id를 가르키고 있어서 데이터를 반환할 때 문제가 없나요?

ByungilOh-Fillip commented:
- [ ] save API 메소드 및 이름 변경
- [ ] 캐싱처리 불 필요한 조회 제거

DongminL committed:
[Feature] 알림 계정별 알림 설정 유무 (#146)

* feat: 알림 허용 여부 컬럼 추가

* feat: 알림을 허용한 사용자에게만 보내도록 조건 추가

* feat: 알림을 설정 변경 API

* feat: Path Valuable 값과 요청한 사람이 다를 경우의 에러 추가

* docs: 알림을 설정 변경 API 명세화

* test: 알림을 설정 변경 테스트

* test: 알림 설정을 해제한 계정으로 알림 테스트

* fix: notifyEnabled 필드 추가 및 호환성 보완

* fix: 요청값 설명 수정

* fix: 피드백 상세 조회 API Filter 통과되도록 설정

ByungilOh-Fillip committed:
refactor: 리뷰 적용 save API 메소드 및 이름 변경
검색 param id로 변경

ByungilOh-Fillip committed:
refactor: 리뷰 적용 캐싱에 불필요한 조회 제거 및 dto 수정

ByungilOh-Fillip committed:
refactor: test 수정 local build 통과

ByungilOh-Fillip committed:
Merge remote-tracking branch 'origin/feature/25-메인-redis에-신호등-정보-저장하기' into feature/25-메인-redis에-신호등-정보-저장하기

# Conflicts:
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CrossroadRedisRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CrossroadRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CustomCrossroadRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CustomCrossroadRepositoryImpl.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/service/CrossroadService.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/dto/TrafficResponse.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepositoryImpl.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/TrafficRedisRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/TrafficRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/service/TrafficService.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/unifiedSignal/controller/SignalController.java
#	src/test/java/org/programmers/signalbuddyfinal/domain/unifiedSignal/controller/SignalControllerTest.java

ByungilOh-Fillip committed:
refactor: save API 메소드 및 이름 변경
검색 param id로 변경

ByungilOh-Fillip committed:
refactor: save API 메소드 및 이름 변경
검색 param id로 변경

ByungilOh-Fillip committed:
refactor: 캐싱처리 불필요한 조회 제거

ByungilOh-Fillip committed:
refactor: 캐싱처리 불필요한 조회 제거

ByungilOh-Fillip committed:
refactor: Test 수정 local build 통과 완료

ByungilOh-Fillip committed:
refactor: Test 수정 local build 통과 완료

ByungilOh-Fillip committed:
Merge remote-tracking branch 'origin/feature/25-메인-redis에-신호등-정보-저장하기' into feature/25-메인-redis에-신호등-정보-저장하기

# Conflicts:
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CrossroadRedisRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CustomCrossroadRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CustomCrossroadRepositoryImpl.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepositoryImpl.java

DongminL committed:
[Feature] 운영 서버 CI/CD 설계 (#151)

* rename: 파일명 변경

* chore: develop(개발 서버용)와 main(운영 서버용)로 도커 이미지 생성 Action 분리

* chore: main 브랜치용 젠킨스 파이프라인

* chore: 개발, 운영 서버 설정 파일 분리

* chore: 개발, 운영 서버 설정 파일 분리

ByungilOh-Fillip commented on src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/service/CrossroadService.java:
```diff
@@ -56,6 +62,66 @@ public void saveCrossroadDates(int page, int pageSize) {
+
+    public CrossroadResponse crossroadFindById(Long id) {
+
+        CrossroadResponse responseRedis = crossroadRedisRepository.findById( id );
+        CrossroadResponse responseDB = new CrossroadResponse(crossroadRepository.findByCrossroadId(id));
```
감사합니다..! 레디스에서 위치 조회하는 기능을 별도로 만들어야했는데... 빨리 수정해서 올려두겠습니다!

DongminL replied:
레디스에서 위치 조회하는 기능은 이미 만드신 거 아닌가요?

ByungilOh-Fillip replied:
제가 위치라고 표현했네요..! 그 반경내에 있는 데이터로 호출하려면 저장을 Redis의 GeoOperation값으로 저장해서 검색해야 DB랑 똑같이 범위 검색 진행할 수 있다고 합니다..! 아니면 애초에 저장할 떄 범위 데이터로 저장하니까 Redis에서도 주변 반경으로 검색하는거 까지는 필요 없을까요?

DongminL replied:
`GeoOperation` 적용하는 건 `Point` 객체 추가 되는 거 말고는 큰 차이가 없는 거 같은데, 오늘이나 내일 낮까지 가능하신가요?

더 오래 걸리시면 교차로 데이터 캐싱은 뒤로 미루는 게 좋을 것 같습니다!

ByungilOh-Fillip replied:
네..! 답장 늦어서 죄송해요 기능 구현 다 했는데 지금 swagger Test 중에 에러가 조금있어서.. 이것만 빨리 해결해보겠습니다!

DongminL committed:
[Fix] yml 파일 값 쌍따옴표로 감싸기 (#157)

fix: 비밀번호 쌍따옴표로 감싸기

Dongjin0224 committed:
[Feature] 날씨 api 기능 구현 (#153)

* chore: 엑셀 파일 읽기 위한 의존성 추가

* feat: WebClient 설정 추가

- lombok.config 추가하여 `@RequiredArgsConstructor`와 `@Qualifier` 같이 쓸 수 있게 설정

* feat: 날씨 API 기능 개발

- 엑셀 데이터 읽고 저장
- SSE 사용자의 위치를 받아서 날씨 1분마다 제공

* test: SSE 테스트

* feat: SSE Security 설정

* fix: 미사용 코드 제거

* fix: 컨벤션 적용

- 임포트, 포맷터 적용

* fix: 파일 경로를 입력받지 않고 내부 경로에서 가져오는거로 변경

* chore: 서브모듈 업데이트

* fix: 테스트 코드 수정

* [Fix] yml 파일 값 쌍따옴표로 감싸기 (#157)

fix: 비밀번호 쌍따옴표로 감싸기

* chore: 엑셀 파일 읽기 위한 의존성 추가

* feat: WebClient 설정 추가

- lombok.config 추가하여 `@RequiredArgsConstructor`와 `@Qualifier` 같이 쓸 수 있게 설정

* feat: 날씨 API 기능 개발

- 엑셀 데이터 읽고 저장
- SSE 사용자의 위치를 받아서 날씨 1분마다 제공

* test: SSE 테스트

* feat: SSE Security 설정

* fix: 미사용 코드 제거

* fix: 컨벤션 적용

- 임포트, 포맷터 적용

* fix: 파일 경로를 입력받지 않고 내부 경로에서 가져오는거로 변경

* fix: 테스트 코드 수정

* test: 날씨 기능 테스트 코드 추가

* feat: 날씨 API 'excludeGetPaths' 추가

---------

Co-authored-by: Lee Dongmin <milkskfk12@gmail.com>

Dongjin0224 committed:
[Featuer] 최근 경로에 필드 추가 (#161)

feat: 최근경로 'address' 필드 추가

Dongjin0224 committed:
[Bug] 최근경로 저장 버그 (#163)

fix: 최근 경로 저장하면 'address' 필드가 저장이 안되는 버그 해결

ByungilOh-Fillip committed:
refactor: submodule 최신화

ByungilOh-Fillip committed:
refactor: 캐싱 기능 정리 및 레디스 범위검색 기능 추가

ByungilOh-Fillip committed:
chore: 사용하지 않는 코드 정리

ByungilOh-Fillip committed:
Merge remote-tracking branch 'origin/feature/25-메인-redis에-신호등-정보-저장하기' into feature/25-메인-redis에-신호등-정보-저장하기

# Conflicts:
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/controller/CrossroadController.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CrossroadRedisRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/service/CrossroadService.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/dto/TrafficResponse.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/CustomTrafficRepositoryImpl.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/repository/TrafficRedisRepository.java
#	src/main/java/org/programmers/signalbuddyfinal/domain/trafficSignal/service/TrafficService.java
#	src/test/java/org/programmers/signalbuddyfinal/domain/unifiedSignal/controller/SignalControllerTest.java

ByungilOh-Fillip committed:
refactor : swagger test 이상 2번째 캐싱 데이터 호출 실패 해결

ByungilOh-Fillip committed:
refactor : SonarCloud 리뷰 반영

ByungilOh-Fillip committed:
refactor : SonarCloud 리뷰 반영2

DongminL commented on src/main/java/org/programmers/signalbuddyfinal/domain/crossroad/repository/CrossroadRedisRepository.java:
```diff
@@ -0,0 +1,123 @@
+            redisTemplate.opsForGeo().add(
+                KEY_GEO,
+                new Point(crossroadResponse.getLng(), crossroadResponse.getLat()),
+                idStr
```
GeoOperation은 CrossroadResponse 객체를 저장하지 못하나요?

ByungilOh-Fillip replied:
네.. geo로 별도로 point나 위도 경도 관리하게 되어있고 이걸로 저장해야 범위 검색이 가능한거 같더라구요..ㅠㅠ 아니면 범위 검색 알고리즘을 double로 처리가 가능하게 만들어야할거 같구요

ByungilOh-Fillip replied:
TTL사라지는것도 확인해봐서 머지하겠습니다! 감사합니다!

DongminL reviewed (approved):
Geo 데이터를 Redis에 저장하고 불러오고, 설정한 TTL 이후로 자동으로 없어지는 것을 확인하셨다면 머지 부탁드립니다!

DongminL commented on src/test/java/org/programmers/signalbuddyfinal/domain/crossroad/service/CrossroadServiceLocationTest.java:
```diff
@@ -0,0 +1,58 @@
+    @Mock
+    private RedisTemplate<Object,Object> redisTemplate;
+
+    @Mock
+    private CrossroadRedisRepository crossroadRedisRepository;
```
단순한 CRUD 테스트는 Mocking 해도 별 문제는 없겠지만, 처음 사용하는 Redis의 GeoOperation을 사용하는 코드를 테스트할 경우에는 Mocking 하지 않는 통합테스트가 어울려 보입니다!

만약 실제 동작을 따로 확인하셨다면 시간 관계상 넘어가는 편이 좋겠습니다!

ByungilOh-Fillip replied:
실제 동작확인했습니다! 오..실제 동작을 확인하는거니까 모킹은 조금 불필요 하긴 하겠네요..! 추후에 변경 꼭 해두겠습니다!
